src/Shared/PropertyHelper/PropertyHelper.cs

```diff
@@ -3,22 +3,20 @@
 
 #nullable enable
 
-using System;
 using System.Collections.Concurrent;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
-using System.Linq;
 using System.Reflection;
 using System.Reflection.Metadata;
 
 [assembly: MetadataUpdateHandler(typeof(Microsoft.Extensions.Internal.PropertyHelper))]
 
 namespace Microsoft.Extensions.Internal;
 
-internal class PropertyHelper
+internal sealed class PropertyHelper
 {
     private const BindingFlags DeclaredOnlyLookup = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static | BindingFlags.DeclaredOnly;
+    private const BindingFlags Everything = BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance;
```

Comment:
We were previously calling [`GetRuntimeProperties`](https://source.dot.net/#System.Private.CoreLib/RuntimeReflectionExtensions.cs,27) inside of a Linq expression which made the trimmer sad. I switched it to call the underlying APIs in a iterative loop.